docs/pages/management/register-services/register-services-join-token.mdx

```diff
@@ -0,0 +1,253 @@
+---
+
+In this guide, we will show you how to register a Teleport service with your
+cluster by presenting a **join token**. 
+
+In this approach, you declare your intention to register a new service, and
+Teleport generates a secure token for the service. When the service starts, it
+presents the token, indicating to your Teleport cluster that it is trustworthy.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- A Linux server that you will use to host your Teleport service. 
+
+  In this guide, we will show you how to register a Teleport SSH Service
+  instance. This approach applies to new Proxy Service instances as well as the
+  Kubernetes Service, Database Service, and other Teleport services for
+  accessing resources in your infrastructure.
+
+(!docs/pages/includes/tctl.mdx!)
+
+## Step 1/3. Install Teleport
+
+Install Teleport on the host where you will run your Teleport service.
+
+(!docs/pages/includes/install-linux.mdx!)
+
+## Step 2/3. Join your service to the cluster
+
+In this section, we will join your Teleport service to the Teleport cluster by:
+
+- Obtaining a join token
+- Running your Teleport service with the join token
+
+### Generate a token
+
+Teleport only allows access to services that have joined the cluster.
+
+Once a service joins, it receives a host certificate signed by the cluster's
+Auth Service. To receive a host certificate upon joining a cluster, a new
+Teleport host must present an **invite token**.
+
+An invite token also defines which role a new host can assume within a cluster:
+
+- `proxy`
+- `node`
+- `auth`
+- `app`
+- `db`
+- `kube`
+- `windowsdesktop`
+
+Administrators can generate tokens as they are needed. A token can be used
+multiple times until its time to live (TTL) expires.
+
+On your local machine, use the `tctl` tool to generate a new token. In the
+following example, a new token is created with a TTL of five minutes:
+
+```code
+# Generate a short-lived invite token for a new node:
+$ export INVITE_TOKEN=$(tctl nodes add --ttl=5m --roles=node | grep "invite token:" | grep -Eo "[0-9a-z]{32}")
+$ echo $INVITE_TOKEN
+# (=presets.tokens.first=)
+
+# You can also list all generated non-expired tokens:
+$ tctl tokens ls
+# Token                            Type            Expiry Time
+# ------------------------         -----------     ---------------
+# (=presets.tokens.first=)         service            25 Sep 18 00:21 UTC
+```
+
+<Details title="Provide your own token value">
+
+Rather than automatically generating a token, you can create one with a known
+value by using the `--token` flag:
+
+```code
+$ tctl nodes add --ttl=5m --roles=node,proxy --token=secret-value
+# The invite token: secret-value
+```
+
+The value of `--token` should be cryptographically secure. For example, you can
+create an SHA256 hash to reduce the risk that a malicious actor will produce
+a token with the same value as yours:
+
+```code
+$ head -n 1 /dev/random | sha256sum
```

Comment:
Let's just remove this section. It's easier to inadvertently create a token without sufficient randomness this way. While this is possible, it's best if the user just let's teleport generate the token.